main/lsp/LSPTask.cc

```diff
@@ -128,4 +243,41 @@ void LSPTask::addLocIfExists(const core::GlobalState &gs, vector<unique_ptr<Loca
     }
 }
 
+LSPQueuePreemptionTask::LSPQueuePreemptionTask(const LSPConfiguration &config, absl::Notification &finished,
+                                               absl::Mutex &taskQueueMutex, TaskQueueState &taskQueue,
+                                               LSPIndexer &indexer)
+    : LSPTask(config, LSPMethod::SorbetError), finished(finished), taskQueueMutex(taskQueueMutex), taskQueue(taskQueue),
+      indexer(indexer) {}
+
+void LSPQueuePreemptionTask::run(LSPTypecheckerDelegate &tc) {
+    for (;;) {
+        unique_ptr<LSPTask> task;
```

Comment:
This is what runs during preemption: A task loop that runs every task at the front of the queue that can preempt.